patches/wlroots/0001-Add-managarm-specific-changes.patch

```diff
@@ -0,0 +1,77 @@
+From 1a8bb5f64f785e2be534e6d79fabdc77c5f322d4 Mon Sep 17 00:00:00 2001
+From: no92 <no92.mail@gmail.com>
+Date: Thu, 27 Jan 2022 16:55:49 +0100
+Subject: [PATCH] Add managarm-specific changes
+
+---
+ include/wlr/backend.h        | 2 ++
+ types/wlr_export_dmabuf_v1.c | 1 +
+ types/wlr_linux_dmabuf_v1.c  | 1 +
+ xwayland/meson.build         | 2 ++
+ xwayland/server.c            | 1 +
+ 5 files changed, 7 insertions(+)
+
+diff --git a/include/wlr/backend.h b/include/wlr/backend.h
+index 025f74b..a7ee826 100644
+--- a/include/wlr/backend.h
++++ b/include/wlr/backend.h
+@@ -12,6 +12,8 @@
+ #include <wayland-server-core.h>
+ #include <wlr/backend/session.h>
+ 
++#include <time.h>
++
+ struct wlr_backend_impl;
+ 
+ struct wlr_backend {
+diff --git a/types/wlr_export_dmabuf_v1.c b/types/wlr_export_dmabuf_v1.c
+index a8e2a4b..0904494 100644
+--- a/types/wlr_export_dmabuf_v1.c
++++ b/types/wlr_export_dmabuf_v1.c
+@@ -1,5 +1,6 @@
+ #include <assert.h>
+ #include <stdlib.h>
++#include <stdio.h>
+ #include <unistd.h>
+ #include <wlr/interfaces/wlr_output.h>
+ #include <wlr/render/dmabuf.h>
+diff --git a/types/wlr_linux_dmabuf_v1.c b/types/wlr_linux_dmabuf_v1.c
+index 7b08b35..4fac6ba 100644
+--- a/types/wlr_linux_dmabuf_v1.c
++++ b/types/wlr_linux_dmabuf_v1.c
+@@ -2,6 +2,7 @@
+ #include <assert.h>
+ #include <drm_fourcc.h>
+ #include <stdlib.h>
++#include <stdio.h>
+ #include <unistd.h>
+ #include <wayland-server-core.h>
+ #include <wlr/render/drm_format_set.h>
+diff --git a/xwayland/meson.build b/xwayland/meson.build
+index b7bbcaf..b7d4013 100644
+--- a/xwayland/meson.build
++++ b/xwayland/meson.build
+@@ -11,6 +11,8 @@ xwayland_optional = {
+ 	'xcb-errors': 'Required for printing X11 errors.',
+ }
+ 
++meson.override_find_program('Xwayland', files('/var/lib/managarm-buildenv/build/system-root/usr/bin/Xwayland'))
```

Comment:
This doesn't work outside of our standard rootfs. We need to fix this in a more general way.

Reply:
why is this even needed? isn't this discovered via .pcs

Reply:
wlroots states that xwayland just doesn't ship `.pc` files, so we need this. Using `sed` to replace a variable seems like the reasonable solution, though.